kube/deployment.go

```diff
@@ -62,23 +62,23 @@ func addAffinityRules(instanceGroup *model.InstanceGroup, spec *helm.Mapping, se
 		if instanceGroup.Run.Affinity.PodAffinity != nil {
 			return errors.New("pod affinity in role manifest not supported")
 		}
+	}
 
-		if settings.CreateHelmChart {
-			podSpec := spec.Get("template", "spec").(*helm.Mapping)
-
-			podSpec.Add("affinity", getAffinityBlock(instanceGroup))
-			podSpec.Sort()
-		}
+	if settings.CreateHelmChart {
+		podSpec := spec.Get("template", "spec").(*helm.Mapping)
 
-		meta := spec.Get("template", "metadata").(*helm.Mapping)
-		if meta.Get("annotations") == nil {
-			meta.Add("annotations", helm.NewMapping())
-			meta.Sort()
-		}
-		annotations := meta.Get("annotations").(*helm.Mapping)
+		podSpec.Add("affinity", getAffinityBlock(instanceGroup))
+		podSpec.Sort()
+	}
 
-		annotations.Sort()
+	meta := spec.Get("template", "metadata").(*helm.Mapping)
+	if meta.Get("annotations") == nil {
+		meta.Add("annotations", helm.NewMapping())
+		meta.Sort()
 	}
+	annotations := meta.Get("annotations").(*helm.Mapping)
+
+	annotations.Sort()
```

Comment:
So this chunk ensures that there's a `.template.metadata.annotations`, which might be empty? Does this actually do anything _useful_?

I recognize this is old code but still…

Reply:
Looking over the code I believe that we could indeed add a `return` after line 77 (`meta.Sort()`) without changing results.

Lines 79-81 are, I believe, needed only if we had annotations when coming into this function. These have to be sorted. When it is us here adding an empty set of annotations, sorting is irrelevant.

Reply:
I confirm that the proposed change passes the fissile unit tests.

Reply:
Looks ok in the SCF chart too. No changes compared to without the change. And the PR as-is inserts the proper affinity templating (affected roles: autoscaler-*, blobstore, credhub).